src/sentry/integrations/bitbucket_server/repository.py

```diff
@@ -0,0 +1,124 @@
+        installation = self.get_installation(data.get("installation"), organization.id)
+        client = installation.get_client()
+
+        try:
+            resp = client.create_hook(
+                data["project"],
+                data["repo"],
+                {
+                    "name": "sentry-bitbucket-server-repo-hook",
+                    "url": absolute_uri(
+                        reverse(
+                            "sentry-extensions-bitbucketserver-webhook",
+                            kwargs={
+                                "organization_id": organization.id,
+                                "integration_id": data.get("installation").encode("utf-8"),
+                            },
+                        )
+                    ),
+                    "active": True,
+                    "events": ["repo:refs_changed", "pr:merged"],
+                },
+            )
+        except Exception as e:
+            installation.raise_error(e)
+        else:
+            return {
+                "name": data["identifier"],
+                "external_id": data["external_id"],
+                "url": installation.model.metadata["base_url"]
+                + u"/projects/{project}/repos/{repo}/browse".format(
+                    project=data["project"], repo=data["repo"]
+                ),
+                "config": {
+                    "name": data["identifier"],
+                    "project": data["project"],
+                    "repo": data["rep"],
+                    "webhook_id": resp["id"],
+                },
+                "integration_id": data["installation"],
+            }
+
+    def on_delete_repository(self, repo):
+        installation = self.get_installation(repo.integration_id, repo.organization_id)
+        client = installation.get_client()
+
+        try:
+            client.delete_hook(
+                repo.config["project"], repo.config["repo"], repo.config["webhook_id"]
+            )
+        except ApiError as exc:
+            if exc.code == 404:
+                return
+            raise
+
+    def _format_commits(self, repo, commit_list):
+        return [
+            {
+                "id": c["id"],
+                "repository": repo.name,
+                "author_email": c["author"]["emailAddress"],
+                "author_name": c["author"]["name"],
+                "message": c["message"],
+                "timestamp": self.format_date(c["authorTimestamp"]),
+                "patch_set": None,
+            }
+            for c in commit_list
+        ]
+
+    def compare_commits(self, repo, start_sha, end_sha):
+        installation = self.get_installation(repo.integration_id, repo.organization_id)
+        client = installation.get_client()
+        # use config name because that is kept in sync via webhooks
+
+        # Bitbucket servers send an empty commit as a string of zeros
+        if "0000000000000000000000000000000000000000" == start_sha:
```

Comment:
@pierredup Could we do something like: 
```
"0"*40
```
I think it's a bit cleaner than having to write out 40 `0`s.